app/controllers/incomeSources/add/ForeignPropertyCheckDetailsController.scala

```diff
@@ -62,54 +65,49 @@ class ForeignPropertyCheckDetailsController @Inject()(val checkForeignPropertyDe
       controllers.incomeSources.add.routes.ForeignPropertyCheckDetailsController.submit()
     }
-
-    if (isDisabled(IncomeSources)) {
-      if (isAgent) Future.successful(Redirect(controllers.routes.HomeController.showAgent)) else Future.successful(Redirect(controllers.routes.HomeController.show()))
-    } else {
-      getDetails(user) map {
-        case Right(viewModel) =>
-          Ok(checkForeignPropertyDetails(
-            viewModel,
-            postAction = postAction,
-            isAgent,
-            backUrl = backUrl
-          ))
-        case Left(ex) =>
-          if (isAgent) {
-            Logger("application").error(
-              s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: ${ex.getMessage}")
-            errorHandler.showInternalServerError()
-          } else {
-            Logger("application").error(
-              s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: ${ex.getMessage}")
-            errorHandler.showInternalServerError()
-          }
-      } recover {
-        case ex: Exception =>
-          if (isAgent) {
-            Logger("application").error(
-              s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
-            errorHandler.showInternalServerError()
-          } else {
-            Logger("application").error(
-              s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
-            errorHandler.showInternalServerError()
-          }
-      }
+    getDetails(user) map {
+      case Right(viewModel) =>
+        Ok(checkForeignPropertyDetails(
+          viewModel,
+          postAction = postAction,
+          isAgent,
+          backUrl = backUrl
+        ))
+      case Left(ex) =>
+        if (isAgent) {
+          Logger("application").error(
+            s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: ${ex.getMessage}")
+          errorHandler.showInternalServerError()
+        } else {
+          Logger("application").error(
+            s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: ${ex.getMessage}")
+          errorHandler.showInternalServerError()
+        }
+    } recover {
+      case ex: Exception =>
+        if (isAgent) {
+          Logger("application").error(
+            s"[Agent][ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
+          errorHandler.showInternalServerError()
+        } else {
+          Logger("application").error(
+            s"[ForeignPropertyCheckDetailsController][handleRequest] - Error: Unable to construct Future ${ex.getMessage}")
+          errorHandler.showInternalServerError()
+        }
     }
   }
 
   def getDetails(implicit user: MtdItUser[_]): Future[Either[Throwable, CheckForeignPropertyViewModel]] = {
```

Comment:
Could fix: this is returning a view model, so it'd be clearer to call it getCheckForeignPropertyViewModel